perfkitbenchmarker/benchmarks/cassandra_stress_benchmark.py

```diff
@@ -174,100 +322,127 @@ def Prepare(benchmark_spec):
   configure = functools.partial(cassandra.Configure, seed_vms=[seed_vm])
   vm_util.RunThreaded(configure, cassandra_vms)
   cassandra.StartCluster(seed_vm, cassandra_vms[1:])
+
   if FLAGS.cassandra_stress_command == USER_COMMAND:
     for vm in vm_dict[CLIENT_GROUP]:
       vm.PushFile(FLAGS.cassandra_stress_profile,
                   TEMP_PROFILE_PATH)
 
+  SetupMetadata(benchmark_spec)
+  PreloadCassandraServer(benchmark_spec)
+
 
 def _ResultFilePath(vm):
   return posixpath.join(vm_util.VM_TMP_DIR,
                         vm.hostname + '.stress_results.txt')
 
 
-def RunTestOnLoader(vm, loader_index, keys_per_vm, data_node_ips):
+def WaitForLoaderToFinish(vm):
+  """Watch loader node and wait for it to finish test.
+
+  Args:
+    vm: The target vm.
+  """
+  result_path = _ResultFilePath(vm)
+  while True:
+    resp, _ = vm.RemoteCommand('tail -n 1 ' + result_path)
+    if re.findall(r'END', resp):
+      break
+    if re.findall(r'FAILURE', resp):
+      vm.PullFile(vm_util.GetTempDir(), result_path)
+      raise errors.Benchmarks.RunError(
+          'cassandra-stress tool failed, check %s for details.'
+          % posixpath.join(vm_util.GetTempDir(),
```

Comment:
result_path is the path to the result file on vm, this is the path of the result file on local machine.